File: CarRacer/app/src/main/java/com/carracer/infrastructure/network/models/Dtos/LeaderboardDto.java
package com.carracer.infrastructure.network.models.Dtos;

import com.google.gson.annotations.SerializedName;

public class LeaderboardDto {
    @SerializedName("id")
    private String id;
    @SerializedName("userLogin")
    private String userLogin;
    @SerializedName("type")
    private String type;
    @SerializedName("durationS")
    private double durationS;
    @SerializedName("peakSpeedKmh")
    private double peakSpeedKmh;
    @SerializedName("distanceMeters")
    private double distanceMeters;
    @SerializedName("measuredAt")
    private long measuredAt;

    public String getId() { return id; }
    public String getUserLogin() { return userLogin; }
    public String getType() { return type; }
    public double getDurationS() { return durationS; }
    public double getPeakSpeedKmh() { return peakSpeedKmh; }
    public double getDistanceMeters() { return distanceMeters; }
    public long getMeasuredAt() { return measuredAt; }
}

File: CarRacerAPI/Dtos/LeaderboardDto.cs
namespace CarRacerAPI.Dtos;

public class LeaderboardDto
{
    public Guid Id { get; set; }
    public string UserLogin { get; set; } = string.Empty; 
    public string Type { get; set; } = string.Empty;
    public double DurationS { get; set; }
    public double PeakSpeedKmh { get; set; }
    public double DistanceMeters { get; set; }
    public long MeasuredAt { get; set; }
}